.github/actions/setup_nix/action.yml

```diff
@@ -22,11 +22,23 @@ runs:
     - name: use btrfs for nix builds
       shell: bash
       run: |
-        sudo mkdir /nixbld
-        truncate -s 3G btrfs.img
-        sudo mkfs.btrfs -f btrfs.img
-        sudo mount btrfs.img /nixbld
+        sudo df -h
+
+        echo "Removing unwanted software... "
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /usr/local/lib/android
+        sudo rm -rf /opt/ghc
+        sudo rm -rf /opt/hostedtoolcache/CodeQL
+        sudo docker system prune --all --force
+        sudo df -h
```

Comment:
I think we might want to drop the `df`s here. Calculating the disk usage takes quite some time

Reply:
The full action takes 20s, I don't think this needs optimization, especially as we likely need further work on it.

Reply:
Maybe you're confusing `df` with `du`?